rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp

```diff
@@ -686,14 +669,214 @@ error_code cellSysutilAvc2SetSpeakerMuting(u8 muting)
+			{
+				streaming_mode = settings.streaming_mode;
+			}
+		}
+		else
+		{
+			switch (init_param->direct_streaming_mode)
+			{
+			case CELL_SYSUTIL_AVC2_STREAMING_MODE_NORMAL:
+				streaming_mode = CELL_SYSUTIL_AVC2_STREAMING_MODE_NORMAL;
+				break;
+			case CELL_SYSUTIL_AVC2_STREAMING_MODE_DIRECT_WAN:
+				break;
+			case CELL_SYSUTIL_AVC2_STREAMING_MODE_DIRECT_LAN:
+				if (init_param->streaming_mode.mode == CELL_SYSUTIL_AVC2_STREAMING_MODE_NORMAL)
+				{
+					settings.streaming_mode = streaming_mode;
+					return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+				}
+				break;
+			default:
+				return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+			}
+		}
+
+		settings.streaming_mode = streaming_mode;
+		settings.mic_out_stream_sharing = init_param->voice_param.mic_out_stream_sharing;
+
+		if (!callback_func)
+		{
+			return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+		}
+
+		if (settings.avc2_cb)
+		{
+			return CELL_AVC2_ERROR_ALREADY_INITIALIZED;
+		}
+
+		settings.avc2_cb = callback_func;
+		settings.avc2_cb_arg = user_data;
+
+		// TODO
+		//return cellSysutilRegisterCallbackDispatcher(0x120, &PTR_LAB_0000552c);
+		break;
+	}
+	case CELL_SYSUTIL_AVC2_VIDEO_CHAT:
+	{
+		if (false) // TODO: syscall to check container
+		{
+			return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+		}
+
+		if (init_param->avc_init_param_version <= 140)
+		{
+			if (false) // TODO
+			{
+				return CELL_AVC2_ERROR_OUT_OF_MEMORY;
+			}
+		}
+
+		if (callback_func)
+		{
+			return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+		}
+
+		bool is_invalid = false;
+
+		if (init_param->video_param.frame_mode == CELL_SYSUTIL_AVC2_FRAME_MODE_NORMAL)
+		{
+			if (init_param->video_param.max_video_windows == 0 || init_param->video_param.max_video_windows > 6)
+			{
+				return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+			}
+		}
+		else
+		{
+			if (init_param->video_param.max_video_windows == 0 || init_param->video_param.max_video_windows > 16)
+			{
+				return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+			}
+		}
+
+		if (init_param->video_param.max_video_bitrate < 1000 || init_param->video_param.max_video_bitrate > 512000)
+		{
+			return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+		}
+
+		if (init_param->video_param.max_video_framerate == 0 || init_param->video_param.max_video_framerate > 30)
+		{
+			return CELL_AVC2_ERROR_INVALID_ARGUMENT;
+		}
+
+		s32 bitrate = 0;
+
+		switch (init_param->video_param.max_video_resolution)
+		{
+		case CELL_SYSUTIL_AVC2_VIDEO_RESOLUTION_QQVGA:
+			bitrate = 76800;
+			break;
+		case CELL_SYSUTIL_AVC2_VIDEO_RESOLUTION_QVGA:
+			bitrate = 307200;
+			break;
+		default:
+			break;
+		}
+
+		u32 unk = 0;
+
+		if (bitrate != 0)
+		{
+			u32 window_count = init_param->video_param.max_video_windows;
+
+			if (init_param->video_param.video_stream_sharing == CELL_SYSUTIL_AVC2_VIDEO_SHARING_MODE_2)
+			{
+				window_count++;
+			}
+
+			const u32 total_bitrate = window_count * bitrate;
+			unk = ((s32(total_bitrate) >> 0x14) + u32(s32(total_bitrate) < 0 && (total_bitrate & 0xfffff) != 0)) * 0x100000 + 0x100000;
```

Comment:
```suggestion
			unk = utils::align<u32>(total_bitrate, 0x100000);
```

Reply:
heh. I had this in a loop and forgot to add the `* 0x100000 + 0x100000` , so it only printed numbers between 0 and 4, so I made it unk and ignored it xD

Reply:
fixed